### ISSUE TITLE
[java] A false negative about UseStringBufferLength 

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3453]
**Affects PMD Version:6.35.0** 

**Rule:UseStringBufferLength**

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd-6.36.0/pmd_rules_java_performance.html#usestringbufferlength

**Description:**
Hi, I found a false negative about the rule UseStringBufferLength. The minimal case is below. Considering this rule, variable str is a constant (final type), I think our tool should have reported a warning at line 4. Thanks. 

**Code Sample demonstrating the issue:**

```java
public void useStringBufferLength() {
   StringBuffer sb = new StringBuffer("Hello PMD!");
   final String str = "";
   sb.toString().equals(str); // should report a warning here
}
```

**Expected outcome: A warning based on this rule**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by tweimer** [comment]
Same as #3639

**Comment by adangel** [comment]
Closing as duplicate of #3639
